pkg/storage/stores/shipper/downloads/table.go

```diff
@@ -100,6 +97,73 @@ func NewTable(spanCtx context.Context, name, cacheLocation string, storageClient
 	return &table
 }
 
+// LoadTable loads a table from local storage(syncs the table too if we have it locally) or downloads it from the shared store.
+func LoadTable(name, cacheLocation string, storageClient StorageClient, boltDBIndexClient BoltDBIndexClient, metrics *metrics) (*Table, error) {
```

Comment:
I think the context should flow here, and the tablemanager.Stop should cancel this.